src/Api/HL/Controller/ITILController.php

```diff
@@ -147,21 +174,107 @@ public static function getRawKnownSchemas(): array
                     'x-readonly' => true,
                 ],
                 'content' => ['type' => Doc\Schema::TYPE_STRING],
+                'is_private' => ['type' => Doc\Schema::TYPE_BOOLEAN],
             ]
         ];
 
+        $schemas['TicketTask'] = $base_task_schema;
+        $schemas['TicketTask']['x-itemtype'] = \TicketTask::class;
+        $schemas['TicketTask']['properties'][Ticket::getForeignKeyField()] = ['type' => Doc\Schema::TYPE_INTEGER, 'format' => Doc\Schema::FORMAT_INTEGER_INT64];
+
+        $schemas['ChangeTask'] = $base_task_schema;
+        $schemas['ChangeTask']['x-itemtype'] = \ChangeTask::class;
+        $schemas['ChangeTask']['properties'][Change::getForeignKeyField()] = ['type' => Doc\Schema::TYPE_INTEGER, 'format' => Doc\Schema::FORMAT_INTEGER_INT64];
+
+        $schemas['ProblemTask'] = $base_task_schema;
+        $schemas['ProblemTask']['x-itemtype'] = \ProblemTask::class;
+        $schemas['ProblemTask']['properties'][Problem::getForeignKeyField()] = ['type' => Doc\Schema::TYPE_INTEGER, 'format' => Doc\Schema::FORMAT_INTEGER_INT64];
+
         $schemas['Followup'] = [
             'type' => Doc\Schema::TYPE_OBJECT,
+            'x-itemtype' => \ITILFollowup::class,
             'properties' => [
                 'id' => [
                     'type' => Doc\Schema::TYPE_INTEGER,
                     'format' => Doc\Schema::FORMAT_INTEGER_INT64,
                     'x-readonly' => true,
                 ],
+                'itemtype' => ['type' => Doc\Schema::TYPE_STRING,],
+                'items_id' => ['type' => Doc\Schema::TYPE_INTEGER, 'format' => Doc\Schema::FORMAT_INTEGER_INT64],
```

Comment:
The x-readonly is not correct for Solution.
From a user point of view, the itemtype and items_id cannot be changed since they are linked to the API path (/Ticket/{id}/...). From the API point though, the fields have to be writable when the item is being created as the controller manually sets these fields while readonly fields are filtered out in at the Search level.